packages/abstract-api/src/endpoints/ibc_callback.rs

```diff
@@ -1,13 +1,64 @@
-use crate::{ApiContract, ApiError};
-use abstract_sdk::{base::endpoints::IbcCallbackEndpoint, AbstractSdkError};
-
-impl<
-        Error: From<cosmwasm_std::StdError> + From<ApiError> + From<AbstractSdkError>,
-        CustomInitMsg,
-        CustomExecMsg,
-        CustomQueryMsg,
-        ReceiveMsg,
-    > IbcCallbackEndpoint
-    for ApiContract<Error, CustomInitMsg, CustomExecMsg, CustomQueryMsg, ReceiveMsg>
+use crate::{state::ContractError, ApiContract};
+use abstract_sdk::base::IbcCallbackEndpoint;
+
+impl<Error: ContractError, CustomInitMsg, CustomExecMsg, CustomQueryMsg, SudoMsg, ReceiveMsg>
+    IbcCallbackEndpoint
+    for ApiContract<Error, CustomInitMsg, CustomExecMsg, CustomQueryMsg, SudoMsg, ReceiveMsg>
 {
 }
+
+// to test this, add ibc client to version control
+
+// #[cfg(test)]
+// mod tests {
+//     use abstract_core::{api::{ExecuteMsg, BaseExecuteMsg, self}, abstract_ica::IbcResponseMsg};
+//     use abstract_sdk::base::ExecuteEndpoint;
+//     use abstract_testing::prelude::{TEST_MANAGER, mocked_account_querier_builder};
+//     use cosmwasm_std::{testing::{mock_dependencies, mock_env, mock_info}, DepsMut, Response};
+//     use speculoos::prelude::*;
+//     use crate::mock::{ApiMockResult, MockReceiveMsg, execute, MOCK_API, mock_init, MockExecMsg, MockError};
```

Comment:
Are we saving this for later or can we remove it?

Reply:
we should save it imo for after IBC revamp

Reply:
We should comment on any commented-out code that with a TODO or FIXME to make sure that we are aware of its purpose.